pkg/controller/bus/controller.go

```diff
@@ -309,46 +309,8 @@ func (c *Controller) syncHandler(key string) error {
 		return err
 	}
 
-	var dispatcherService *corev1.Service
-	var dispatcherDeployment, provisionerDeployment *appsv1.Deployment
-	var dispatcherServiceErr, dispatcherDeplErr, provisionerDeplError error
+	err = c.updateBusStatus(bus)
```

Comment:
updateBusStatus should only be concerned with updating the status field on the bus resource. With your change it is also syncing deployments and services.

Reply:
My intention is to do both in the function, I can probably rename it to `syncAndUpdateBusStatus` or remove the function and add the code into `syncHandler` itself.

Reply:
Do you have other recommendations?

Reply:
the purpose of `syncHandler` is to sync the actual and desired state for the Bus, so it makes sense that it should delegate the syncing of controlled resources and then collect and update the status for the bus based on the results of the syncing.

Reply:
Added them under `syncHandler`